src/failure_message.rs

```diff
@@ -0,0 +1,49 @@
+use lazy_static::lazy_static;
+use phf::phf_ordered_map;
+
+use pyo3::prelude::*;
+
+use itertools::Itertools;
+use regex::Regex;
+
+// Need to use an ordered map to make sure we replace '>' before
+// we replace '\n', so that we don't replace the '>' in '<br>'
+static REPLACEMENTS: phf::OrderedMap<&'static str, &'static str> = phf_ordered_map! {
+    "\"" => "&quot;",
+    "'" => "&apos;",
+    "<" => "&lt;",
+    ">" => "&gt;",
+    "?" => "&amp;",
+    "\r" =>  "",
+    "\n" =>  "<br>",
+};
+
+#[pyfunction]
+pub fn escape_failure_message(mut failure_message: String) -> String {
+    for (from, to) in REPLACEMENTS.entries() {
+        failure_message = failure_message.replace(from, to);
+    }
+    failure_message
+}
+
+lazy_static! {
+    static ref SHORTEN_PATH_PATTERN: Regex =
+        Regex::new(r"(?:\/*[\w\-]+\/)+(?:[\w\.]+)(?::\d+:\d+)*").unwrap();
+}
+
+#[pyfunction]
+pub fn shorten_file_paths(mut failure_message: String) -> String {
+    let original_failure_message = failure_message.clone();
```

Comment:
[nit] Wouldn't it make more sense to have the argument be immutable (which is generally a good idea) and have the local var be `mut` (and return that at the end)?